### ISSUE TITLE
how to Detect  leader changed？

### ISSUE THREAD

**Issue by xiatianqiuyu** [lni/dragonboat#300]
How to check whether the leader has changed? How to register an observer and perform certain operations when the leader sends a change

**Comment by kevburnsjr** [comment]
`config.NodeHostConfig.RaftEventListener`
https://pkg.go.dev/github.com/lni/dragonboat/v4@v4.0.0-20230202152124-023bafb8e648/config#NodeHostConfig
```go
// RaftEventListener is the listener for Raft events, such as Raft leadership
// change, exposed to user space. NodeHost uses a single dedicated goroutine
// to invoke all RaftEventListener methods one by one, CPU intensive or IO
// related procedures that can cause long delays should be offloaded to worker
// goroutines managed by users. See the raftio.IRaftEventListener definition
// for more details.
RaftEventListener raftio.IRaftEventListener
```
```go
type IRaftEventListener interface {
	LeaderUpdated(info LeaderInfo)
}
```
```go
type LeaderInfo struct {
	ShardID   uint64
	ReplicaID uint64
	Term      uint64
	LeaderID  uint64
}
```

**Comment by xiatianqiuyu** [comment]
this method is very good。How to check whether the raft node Breakdowns? How to register an observer and perform certain operations when the node Breakdowns

**Comment by xiatianqiuyu** [comment]
> this method is very good。How to check whether the raft node Breakdowns? How to register an observer and perform certain operations when the node Breakdowns

**Comment by lni** [comment]
your new question is being addressed in #301.

**Comment by paulm17** [comment]
I've got a use case where I have a scheduler.  It only needs to be run on one node.  However I'm using dragonboat to provide HA.

What would be the best mechanism to only ensure the scheduler runs in 1 node only?

Thanks
